dbt/adapters/databricks/connections.py

```diff
@@ -478,51 +478,38 @@ def execute(self, sql: str, bindings: Optional[Sequence[Any]] = None) -> None:
 
         # if the command was to refresh a materialized view we need to poll
         # the pipeline until the refresh is finished.
-        refresh_search = re.search(r"refresh\s+materialized\s+view\s+([`\w.]+)", sql)
-        if not refresh_search:
-            refresh_search = re.search(
-                r"create\s+or\s+refresh\s+streaming\s+table\s+([`\w.]+)", sql
-            )
-        if refresh_search:
-            view_name = refresh_search.group(1).replace("`", "")
-            self.pollMaterializedViewPipeline(view_name)
+        self.pollRefreshPipeline(sql)
 
-    def pollMaterializedViewPipeline(
+    def pollRefreshPipeline(
         self,
-        view_name: str,
+        sql: str,
     ) -> None:
+        should_poll, model_name = _should_poll_refresh(sql)
+        if not should_poll:
+            return
+
         # interval in seconds
         polling_interval = 10
 
         # timeout in seconds
-        timeout = 60 * 60 * 4
+        timeout = 60 * 60
 
         stopped_states = ("COMPLETED", "FAILED", "CANCELED")
-        host = self._creds.host
+        host: str = self._creds.host or ""
         headers = self._cursor.connection.thrift_backend._auth_provider._header_factory()
         headers["User-Agent"] = self._user_agent
 
-        url = f"https://{host}/api/2.1/unity-catalog/tables/{view_name}"
-        resp1 = requests.get(url, headers=headers)
-        if resp1.status_code != 200:
-            raise dbt.exceptions.DbtRuntimeError(
-                f"Error getting materialized view info: {view_name}"
-            )
-
-        pipeline_id = resp1.json().get("pipeline_id")
-        url = f"https://{host}/api/2.0/pipelines/{pipeline_id}"
+        pipeline_id = _get_table_view_pipeline_id(host, headers, model_name)
+        pipeline = _get_pipeline_state(host, headers, pipeline_id)
+        latest_update = _find_update(pipeline)
```

Comment:
Should this variable be called `update` instead to unify this call with lines 524-525? I'm not clear the difference between `update` and `latest_update`.